test/new-e2e/agent-subcommands/secret_test.go

```diff
@@ -0,0 +1,66 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package agentsubcommands
+
+import (
+	"testing"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type agentSecretSuite struct {
+	e2e.Suite[e2e.AgentEnv]
+}
+
+func TestAgentSecretSuite(t *testing.T) {
+	e2e.Run(t, &agentSecretSuite{}, e2e.AgentStackDef(nil))
+}
+
+func (v *agentSecretSuite) TestAgentSecretNotEnabledByDefault() {
+	v.UpdateEnv(e2e.AgentStackDef(nil))
```

Comment:
This is optional as this is the same as https://github.com/DataDog/datadog-agent/pull/18771/files#diff-2e3065c0eee76814c7ed58ba65f14a068e7d4f9d5bf16199b550cf32e2a0933fR22.